Assets/HoloToolkit-Examples/UX/Scripts/InteractiveGroup.cs

```diff
@@ -0,0 +1,137 @@
+                    "please attach one of the Interactive scripts " +
+                    "to your InteractiveGroup Prefab!");
+            }
+            else
+            {
+                UpdateData();
+            }
+        }
+
+        private List<InteractiveToggle> Interactives {
+            get 
+            {
+                return GetInteractiveSet().Interactives;
+            }
+            set 
+            {
+                GetInteractiveSet().Interactives = value;
+            }
+        }
+
+        public InteractiveSet GetInteractiveSet()
+        {
+            return GetComponent<InteractiveSet>();
+        }
+
+        /// <summary>
+        /// create new Prefab-instance and fill with given data
+        /// </summary>
+        void CreateInteractives()
+        {
+            for (int i = Interactives.Count; i < Titles.Count; i++)
+            {
+                GameObject PrefabInst = Instantiate(InteractivePrefab, gameObject.transform) as GameObject;
+                InteractiveToggle InterInst = PrefabInst.GetComponent<InteractiveToggle>();
+                Interactives.Add(InterInst);
+            }
+        }
+
+        /// <summary>
+        /// shortcut to set title
+        /// (assuming this Interactive has a LabelTheme attached to it)
+        /// </summary>
+        /// <param name="title"></param>
+        public void SetTitle(string title)
+        {
+            LabelTheme lblTheme = gameObject.GetComponent<LabelTheme>();
+            if (lblTheme == null)
+            {
+                Debug.LogError("No LabelTheme attached to this Interactive");
+            }
+            else
+            {
+                lblTheme.Default = title;
+            }
+        }
+
+        /// <summary>
+        /// position interactives and set title text.
+        /// </summary>
+        public void UpdateData()
+        {
+            RemoveInteractives(Titles.Count);
+            CreateInteractives();
+
+            int rows = System.Math.Min(Rows, Titles.Count);
+            int columns = (Titles.Count - 1) / Rows + 1;
+
+            for (int i = 0; i < Interactives.Count; i++)
+            {
+                // set title
+                string title = Titles[i];
+                Interactive interactive = Interactives[i];
+                interactive.SetTitle(title);
+                interactive.Keyword = title;
+                
+                // layouting
+                int j = i % rows;
+                Collider collider = interactive.gameObject.GetComponent<Collider>();
+                Vector2 Distance = new Vector2(
+                    collider.bounds.size.x + Offsets.x,
+                    collider.bounds.size.y + Offsets.y
+                );
+                interactive.gameObject.transform.localPosition = new Vector3(
+                    ((i / rows) - ((columns - 1) / 2f)) * Distance.x,
+                    -(j - (rows - 1) / 2f) * Distance.y);
```

Comment:
Instad of `value / 2` we can use `value * 0.5f`

Reply:
I think value / 2 is better for (human) readability and I don't think the CPU will mind (its not like we are doing this loop for thousands of objects). Or is there a coding convention I am missing?

Reply:
Idk, as a human, I've always hated fractions.

as for CPU perf, gotta remember we're also targeting HoloLens, and it's a mobile device and every little bit counts.